### PR TITLE
Javascript loops practice -Daniel Cho

### PR DESCRIPTION
**Pull request by danielycho**
javascript-loops-practice screenshot

### CODE REVIEW

File: javascript-loops-practice/filter-out-nulls.js [hunk 1]
```diff
@@ -0,0 +1,11 @@
+/* exported filterOutNulls */
+function filterOutNulls(values) {
+  var array = [];
+  for (var i = 0; i < values.length; i++) {
+    if (values[i] === null) {
+      values.splice(i, 0);
```
Comment (codingcodymiller) [inline javascript-loops-practice/filter-out-nulls.js:6]: Oh no, this is changing the original array. If the current value is null, just ignore it, don't delete it from the argument array

File: javascript-loops-practice/filter-out-strings.js [hunk 2]
```diff
@@ -0,0 +1,11 @@
+/* exported filterOutStrings */
+function filterOutStrings(values) {
+  var array = [];
+  for (var i = 0; i < values.length; i++) {
+    if (typeof values[i] === 'number') {
+      array.push(values[i]);
+    } else {
+      values.splice(i, 0);
```
Comment (codingcodymiller) [inline javascript-loops-practice/filter-out-strings.js:8]: Same problem here, don't change the original array

File: javascript-loops-practice/filter-out-strings.js [hunk 3]
```diff
@@ -0,0 +1,11 @@
+/* exported filterOutStrings */
+function filterOutStrings(values) {
+  var array = [];
+  for (var i = 0; i < values.length; i++) {
+    if (typeof values[i] === 'number') {
+      array.push(values[i]);
```
Comment (codingcodymiller) [inline javascript-loops-practice/filter-out-strings.js:6]: This function is meant to filter out strings, not collect all numbers. What if the argument also included other data types like` null` or boolean values?

File: javascript-loops-practice/odd-or-even.js [hunk 4]
```diff
@@ -0,0 +1,13 @@
+/* exported oddOrEven */
+function oddOrEven(numbers) {
+  var array = [];
+  for (var i = 0; i < numbers.length; i++) {
+    if (numbers[i] % 2 === 0) {
+      array.push('even');
+    }
+    if (numbers[i] % 2 !== 0) {
```
Comment (codingcodymiller) [inline javascript-loops-practice/odd-or-even.js:8]: This can be an `else` statement for the above `if` statement

**Review by codingcodymiller — CHANGES_REQUESTED** [review]

**Comment by danielycho** [comment]
filteroutstrings
filteroutnulls
oddoreven

File: javascript-loops-practice/filter-out-strings.js [hunk 5]
```diff
@@ -2,10 +2,8 @@
 function filterOutStrings(values) {
   var array = [];
   for (var i = 0; i < values.length; i++) {
-    if (typeof values[i] === 'number') {
+    if (typeof values[i] === 'number' || typeof values[i] === 'boolean' || typeof values[i] === 'object') {
```
Comment (codingcodymiller) [inline javascript-loops-practice/filter-out-strings.js:5]: How about `!== "string"`?

**Review by codingcodymiller — CHANGES_REQUESTED** [review]

**Comment by danielycho** [comment]
image

That way of thinking totally blanked my mind. lol

**Review by codingcodymiller — APPROVED** [review]

**Pull request merged**